cpp/core/shuffle/Payload.cc

```diff
@@ -310,7 +310,7 @@ arrow::Result<std::vector<std::shared_ptr<arrow::Buffer>>> BlockPayload::deseria
   RETURN_NOT_OK(inputStream->Read(sizeof(uint32_t), &numRows));
   uint32_t numBuffers;
   RETURN_NOT_OK(inputStream->Read(sizeof(uint32_t), &numBuffers));
-  timer.reset();
+  deserializeTime += timer.realTimeUsed();
```

Comment:
Why do we need this change? timer.start() and timer.stop() is not called. the realTimeUsed would be zero

Reply:
@marin-ma  Thanks, Um.. I've added this to fix the compile error with `std::unique_ptr<ScopedTimer>`. It was a typo. Now, I have fixed it in another way.